src/__tests__/components/ui4/BlogCards.test.tsx

```diff
@@ -0,0 +1,82 @@
+import { describe, expect, it } from '@jest/globals'
+import { asBlogPostsGeo, BlogPostGeo } from 'edge-info-server'
+
+import { filterBlogCards } from '../../../components/ui4/BlogCards'
+
+describe('filterBlogCards', () => {
+  const dummyData: BlogPostGeo = {
+    localeTitle: {
+      en: 'Title'
+    },
+    localeBody: {
+      en: 'My blog body text'
+    },
+    localeBlogUrl: {
+      en: 'https://edge.app/whatisthorchain'
+    },
+    lightImageUrl: 'https://edge.app/images/tc-dark.png',
+    darkImageUrl: 'https://edge.app/images/tc-light.png',
+    countryCodes: [],
+    excludeCountryCodes: []
+  }
+
+  it('should return empty array if no countryCode provided', () => {
+    expect(filterBlogCards(asBlogPostsGeo([dummyData, dummyData]))).toEqual([])
```

Comment:
If the cards do not require any countryCodes (which should be optional), then a missing countryCode arg should not remove all cards. Check if the card's countryCodes == null or is empty array